### ISSUE TITLE
Bug in grad for jax.nn.log_sigmoid

### ISSUE THREAD

**Issue by nsavinov** [google/jax#2140]
Grad at 0 produces 0 instead of 0.5:

```python
import jax
from jax.test_util import check_grads
import jax.numpy as jnp

print(jax.grad(jax.nn.log_sigmoid)(0.0))
print(jax.grad(lambda x: -jnp.log(1.0 + jnp.exp(-x)))(0.0))
check_grads(jax.nn.log_sigmoid, (0.0,), order=1)
```
```
0.0
0.5
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-37-f31cb0a2731e> in <module>()
     13 print(jax.grad(jax.nn.log_sigmoid)(0.0))
     14 print(jax.grad(lambda x: -jnp.log(1.0 + jnp.exp(-x)))(0.0))
---> 15 check_grads(jax.nn.log_sigmoid, (0.0,), order=1)

9 frames
google3/third_party/py/numpy/testing/_private/utils.py in assert_array_compare(comparison, x, y, err_msg, verbose, header, precision, equal_nan, equal_inf)
    825                                 verbose=verbose, header=header,
    826                                 names=('x', 'y'), precision=precision)
--> 827             raise AssertionError(msg)
    828     except ValueError:
    829         import traceback

AssertionError: 
Not equal to tolerance rtol=0.002, atol=0.002

Mismatch: 100%
Max absolute difference: 0.88214874
Max relative difference: 1.
 x: array(0., dtype=float32)
 y: array(0.882149, dtype=float32)
```

**Comment by nsavinov** [comment]
Also, the following functions fail check_grads at 0:
```python
import jax
from jax.test_util import check_grads
import jax.numpy as jnp

for f in [jax.nn.log_sigmoid,
          jax.nn.sigmoid,
          jax.nn.softplus,
          jnp.exp,
          lambda x: 1.0 / (1.0 + jnp.exp(-x)),
          lambda x: -jnp.log(1.0 + jnp.exp(-x))]:
  try:
    check_grads(f, (0.0,), order=1)
  except:
    print('Grad check failed for', f)
```
```
Grad check failed for <function log_sigmoid at 0x7f6ad28c12f0>
Grad check failed for <function sigmoid at 0x7f6ad28c11e0>
Grad check failed for <function softplus at 0x7f6ad28c10d0>
Grad check failed for <function _one_to_one_unop.<locals>.fn at 0x7f6ad5396730>
Grad check failed for <function <lambda> at 0x7f6a648496a8>
Grad check failed for <function <lambda> at 0x7f6a64849510>
```

**Comment by joaogui1** [comment]
There seems to be a problem when computing the grad of lax.abs too, which may help explain the bug in the softplus and log_sigmoid gradients. 
Also I ran @nsavinov  code on 0.1.57 and only those two functions failed, what version of JAX are you using?

**Comment by nsavinov** [comment]
I also use 0.1.57. But I ran it on TPU internally at google -- not sure if it makes a difference.

**Comment by shoyer** [comment]
See also https://github.com/google/jax/issues/2107

**Comment by alimuldal** [comment]
This should probably be closed now that #2195 is merged

**Comment by nsavinov** [comment]
To be more precise, #2698 is merged, which is a successor to #2195 (from what I understood).
Now `jax.grad(jax.nn.log_sigmoid)(0.0) == 0.5`, so it is correct and I'm closing this bug.
